procedural/timeline/src/main/kotlin/gov/nasa/ammos/aerie/procedural/timeline/plan/Plan.kt

```diff
@@ -41,4 +42,13 @@ interface Plan {
    * @param name string name of the resource
    */
   fun <V: Any, TL: SerialSegmentOps<V, TL>> resource(name: String, deserializer: (List<Segment<SerializedValue>>) -> TL): TL
+
+  /** Get external events associated with this plan. */
+  fun events(query: EventQuery): ExternalEvents
```

Comment:
I'm not quite seeing the purpose of `EventQuery`, since we already provide filtering capabilities in these other event function overloads. Why not just have a function in the `Plan` interface here with nullable arguments (`derivationGroup: String?, eventType: String?, source: String?`) that conditionally handles the filtering?

Not a big deal to me, but it does seem a little confusing that we're exposing multiple filtering APIs to users, especially when they could do this filtering in "userspace" anyways.

Reply:
The only thing this provides in addition is the ability to specify multiple filters in one single query, without needing multiple filtering function calls. They are be functionally equivalent, but, at least in testing of the feature, this query did seem to be pretty helpful. 

@JoelCourtney Any thoughts on this? Should we remove either the `EventQuery` or the other function options?

Reply:
Gotcha, I just worry about exposing too many filtering objects to end users. It looks like we can filter when requesting events in two ways:
1. `val events = plan.events("SomeGroupName")
2. `val events = plan.events(new EventQuery("SomeGroupName"))

And once we have an `ExternalEvents` object, we can filter in two ways:
1. `val justFooType = events.filterByType("Foo")
2. `val justFooType = events.filter { it.type == "Foo }

Like you said, all are functionally equivalent, but I can imagine a user being a bit confused when they see multiple ways to filter events without realizing they're functionally equivalent.

Now (while introducing this feature) is a good time to think about the most clear / concise API we can expose, since future changes would (likely?) be breaking.

Reply:
Actually wait - I think I change my mind. I think we should actually allow all of these. The point of the methods on `Plan`, despite the fact that they can be replicated in `EventQuery`, is to provide idiomatic ways of filtering events from a plan without needing to specify `EventQuery`. We will add more to events (like metadata) in the future and as such having both an easy idiomatic way as well as a more complicated, query way, to fetch events I would argue is helpful. At the current moment there is complete overlap but going forward I don't think this will be the case!

Reply:
Gotcha! That's why I had originally asked about the point of `EventQuery` since it (currently) feels like an unnecessary layer of indirection around simple filtering logic, but if you see the `EventQuery` object encapsulating logic around more complex filters (e.g. some sort of metadata many-to-one mapping or something), then I'm fine keeping it in.

Reply:
Actually EventQuery and the other `events` overloads could provide some actual utility, and the explicit filter functions are just sugar. I made EventQuery to give the Plan implementation the option to optimize the events database lookup so that it only loads the events you need (although the built in scheduler doesn't actually do this).

I think there are situations that are best suited to each of the three ways of filtering, but I agree that having three ways is kinda overkill. So I'd be happy removing the `events(derivGroup)` and `events(group, type)` overloads, since I think they're the least useful.

Reply:
These are the ones defined as methods on `Plan`, right?